File: Java/206_ReverseLinkedList.java
// 206 - Reverse Linked List

public ListNode reverseList(ListNode head) {
    if(head == null) return null;

    ListNode next = null, node = null;
    while(head != null) {
        next = head.next;
        head.next = node;
        node = head;
        head = next;
    }

    return node;
}

File: CPP/206_ReverseLinkedList.cpp
// 206 - Reverse Linked List

ListNode* reverseList(ListNode* head) {
    if(head == nullptr) return head;

    ListNode* next = nullptr;
    ListNode* node = nullptr;
    while(head != nullptr) {
        next = head->next;
        head->next = node;
        node = head;
        head = next;
    }

    return node;
}